src/Ocelot/Middleware/Multiplexer/SimpleJsonResponseAggregator.cs

```diff
@@ -21,19 +22,54 @@ private static async Task MapAggregateContent(DownstreamContext originalContext,
 
             contentBuilder.Append("{");
 
-            for (var i = 0; i < downstreamContexts.Count; i++)
+            var responceKeys = downstreamContexts.Select(s => s.DownstreamReRoute.Key).Distinct().ToList();
+
+            for (var k = 0; k < responceKeys.Count; k++)
             {
-                if (downstreamContexts[i].IsError)
+                var contexts = downstreamContexts.Where(w => w.DownstreamReRoute.Key == responceKeys[k]).ToList();
+                if (contexts.Count == 1)
                 {
-                    MapAggregateError(originalContext, downstreamContexts, i);
-                    return;
+                    if (contexts[0].IsError)
+                    {
+                        MapAggregateError(originalContext, contexts[0]);
+                        return;
+                    }
+
+                    var content = await contexts[0].DownstreamResponse.Content.ReadAsStringAsync();
+                    contentBuilder.Append($"\"{responceKeys[k]}\":{content}");
+
                 }
+                else
+                {
+                    contentBuilder.Append($"\"{responceKeys[k]}\":");
+                    contentBuilder.Append("[");
 
-                var content = await downstreamContexts[i].DownstreamResponse.Content.ReadAsStringAsync();
+                    for (var i = 0; i < contexts.Count; i++)
+                    {
+                        if (contexts[i].IsError)
+                        {
+                            MapAggregateError(originalContext, contexts[i]);
+                            return;
+                        }
 
-                contentBuilder.Append($"\"{downstreamContexts[i].DownstreamReRoute.Key}\":{content}");
+                        var content = await contexts[i].DownstreamResponse.Content.ReadAsStringAsync();
+                        if (string.IsNullOrWhiteSpace(content))
+                        {
+                            continue;
+                        }
+
+                        contentBuilder.Append($"{content}");
+
+                        if (i + 1 < contexts.Count)
+                        {
+                            contentBuilder.Append(",");
+                        }
+                    }
+
+                    contentBuilder.Append("]");
+                }
 
-                if (i + 1 < downstreamContexts.Count)
+                if (k + 1 < responceKeys.Count)
```

Comment:
this is spelled wrong think it should be responseKeys, I dont think k is a good name for a variable.